python/build-wheel-macos.sh

```diff
@@ -16,19 +16,21 @@ DOWNLOAD_DIR=python_downloads
 NODE_VERSION="14"
 PY_VERSIONS=("3.6.1"
              "3.7.0"
-             "3.8.2")
+             "3.8.2"
+             "3.9.0")
 PY_INSTS=("python-3.6.1-macosx10.6.pkg"
           "python-3.7.0-macosx10.6.pkg"
-          "python-3.8.2-macosx10.9.pkg")
+          "python-3.8.2-macosx10.9.pkg"
+          "python-3.9.0-macosx10.9.pkg")
 PY_MMS=("3.6"
         "3.7"
-        "3.8")
+        "3.8"
+        "3.9")
 
-# The minimum supported numpy version is 1.14, see
-# https://issues.apache.org/jira/browse/ARROW-3141
-NUMPY_VERSIONS=("1.14.5"
-                "1.14.5"
-                "1.14.5")
+NUMPY_VERSIONS=("1.19.3"
```

Comment:
Should only bump it for 3.9, right? (same with linux)

Reply:
The tests pass with both, its up to preference imo with what we test against.
I can change it to 1.16 for the older versions if you want.

Reply:
Can we leave the versions at 1.14.5 (what they were before)? 

It seems like supporting the older version will cause strictly less dependency problems for users.

Reply:
Done!